api/src/main/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAO.java

```diff
@@ -100,18 +104,65 @@ public Diagnosis saveDiagnosis(Diagnosis diagnosis) {
 	}
 
 	/**
-	 * Gets primary diagnoses for a given encounter
+	 * Gets diagnoses for an Encounter. When specified, this method only returns
+	 * primary or confirmed diagnoses.
 	 *
-	 * @param encounter the specific encounter to get the primary diagnoses for.
-	 * @return list of primary diagnoses for an encounter
+	 * @param encounter     the encounter for which to fetch diagnoses
+	 * @param primaryOnly   whether to return only primary diagnoses
+	 * @param confirmedOnly whether to return only confirmed diagnoses
+	 * @return the list of (primary, confirmed) diagnoses for the given encounter
 	 */
 	@Override
-	public List<Diagnosis> getPrimaryDiagnoses(Encounter encounter) {
-		Query query = sessionFactory.getCurrentSession().createQuery(
-			"from Diagnosis d where d.encounter.encounterId = :encounterId and d.rank = :rankId order by dateCreated desc");
-		query.setInteger("encounterId", encounter.getId());
-		query.setInteger("rankId", PRIMARY_RANK);
-		return query.list();
+	public List<Diagnosis> getDiagnosesByEncounter(Encounter encounter, boolean primaryOnly, boolean confirmedOnly) {
+		String queryString = "from Diagnosis d where d.encounter.id = :encounterId";
+		if (primaryOnly) {
+			queryString += " and d.rank = :rankId";
+		}
+		if (confirmedOnly) {
+			queryString += " and d.certainty = :certainty";
+		}
+		queryString += " order by d.dateCreated desc";
+
+		TypedQuery<Diagnosis> query = sessionFactory.getCurrentSession().createQuery(queryString, Diagnosis.class).setParameter("encounterId", encounter.getId());
+		if (primaryOnly) {
+			query.setParameter("rankId", PRIMARY_RANK);
+		}
+		if (confirmedOnly) {
+			query.setParameter("certainty", ConditionVerificationStatus.CONFIRMED);
+		}
+
+		return query.getResultList();
+	}
+
+	/**
+	 * Gets diagnoses for an Encounter. When specified, this method only returns
```

Comment:
Did you forget about the `@see` annotation?

Reply:
Yep, I did indeed - apologies.